[nativert] Move file_util to pytorch core

georgiaphillips opened the pull request:
Summary: fbcode//sigmoid/core/common -> fbcode//caffe2/torch/nativert/common

Test Plan: Github CI

Differential Revision: D74328089

zhxchen17 commented:
@georgiaphillips For the fmt header issue, Shangdi has the fix in https://github.com/pytorch/pytorch/pull/152953 and you can wait for that to be landed.

Skylion007 commented on torch/nativert/common/FileUtil.cpp:
```diff
@@ -0,0 +1,206 @@
+  other.release();
+}
+
+File& File::operator=(File&& other) {
```
Why isn't this noexcept? Does release throw or something?

Skylion007 commented on torch/nativert/common/FileUtil.cpp:
```diff
@@ -0,0 +1,206 @@
+  }
+}
+
+bool File::closeNoThrow() {
```
Shouldn't the error value here always be checked? Do we want a [nodiscard] annotation here?

Skylion007 commented on torch/nativert/common/FileUtil.cpp:
```diff
@@ -0,0 +1,206 @@
+}
+
+File& File::operator=(File&& other) {
+  closeNoThrow();
```
Was the value of this supposed to be checked and throw?

Skylion007 commented on torch/nativert/common/FileUtil.cpp:
```diff
@@ -0,0 +1,206 @@
+  }
+}
+
+/* static */ File File::temporary() {
```
Why isn't this static?

cyyever commented:
It's possible to use std::filesystem

zhxchen17 commented:
@georgiaphillips I guess we can bypass the const cast linting with comments and fix the .data() issue by using std::string instead.

georgiaphillips committed:
[nativert] Move file_util to pytorch core (#153162)

Summary:

fbcode//sigmoid/core/common -> fbcode//caffe2/torch/nativert/common

Test Plan: Github CI

Reviewed By: dolpm

Differential Revision: D74328089
